### ISSUE TITLE
Buildcraft 1.7.10 Pumps water consumption

### ISSUE THREAD

**Issue by PiotrKFtw** [BuildCraft/BuildCraft#4545]
Is there a way to make the pumps consume water regardless of the water pool size?
If not, consider this an idea instead of a question.
Note. If not, please don't deny this idea, with a very powerful kinesis engine power generation it's possible to make the pump, pump all the water out, even from a large river. In very old buildcraft versions, it was possible to do and i was doing it to empty some rivers and stuff..

**Comment by AlexIIL** [comment]
Yes, you can change the config option `pumpsConsumeWater` to true.

**Comment by PiotrKFtw** [comment]
> Yes, you can change the config option `pumpsConsumeWater` to true.

So it does actually make it consume water everywhere it can? I couldn't notice water disappearing, even with 25x combustion engine

**Comment by AlexIIL** [comment]
With that option set to true it doesn't treat water any differently than any other fluid - so it sounds like you haven't set the config option correctly?

**Comment by AlexIIL** [comment]
`1.7.10`

Sorry, I didn't read that part, I'll have to re-test there

**Comment by AlexIIL** [comment]
From looking at the code it looks like it should work in the same way there - although I'm just re-testing. (Also, what BC version are you using?)

**Comment by PiotrKFtw** [comment]
> From looking at the code it looks like it should work in the same way there - although I'm just re-testing. (Also, what BC version are you using?)

Latest available.

**Comment by AlexIIL** [comment]
Ok. I think this doesn't work because the pump only pumps fluids out every 16 ticks - which might be too slow to prevent water from creating a new source block. (Unlike 1.12.2 BC which can pump every tick).

**Comment by PiotrKFtw** [comment]
> Ok. I think this doesn't work because the pump only pumps fluids out every 16 ticks - which might be too slow to prevent water from creating a new source block. (Unlike 1.12.2 BC which can pump every tick).

So can you consider this issue as an idea?

**Comment by SandGrainOne** [comment]
Minecraft water flow and the mechanism that creates new water blocks can fail if the player is too far away. Without chunk loaders a chunk can be loaded, but partly inactive. Crops stops growing and water stops flowing, but tile entities like the Pump continue to work. 

It was then decided to change the pump to not remove the water from the world if the pool was 9 tiles or larger. The config option `pumpsConsumeWater` was added later to support mods like Finite Water.

**Comment by PiotrKFtw** [comment]
> Minecraft water flow and the mechanism that creates new water blocks can fail if the player is too far away. Without chunk loaders a chunk can be loaded, but partly inactive. Crops stops growing and water stops flowing, but tile entities like the Pump continue to work.
> 
> It was then decided to change the pump to not remove the water from the world if the pool was 9 tiles or larger. The config option `pumpsConsumeWater` was added later to support mods like Finite Water.

It would be cool if there was option to make it force consume water regardless of the chunk loading or make it load a limited amount of chunks (like 10 or 20)

**Comment by AEnterprise** [comment]
i think you are misunderstanding, if you enable that option it will do exactly that, use water no matter what.

however what is probably happening in your case (though hard to know for sure without pictures) is that the regular minecraft mechanics are kicking in and creating source blocks faster then the pump can use em

**Comment by asiekierka** [comment]
For future readers: If you want to change Minecraft's default water flow mechanics, do exactly that; I don't see why BuildCraft should be special-cased, plus it would take disproportionate amounts of development effort.

**Comment by PiotrKFtw** [comment]
I know it's 4 years later but afaik 

>  if the pool was 9 tiles or larger

Was the problem, adding to configuration option to customize this limit or the possibility to set it to unlimited would do, as well as the fact if it should consume water only if the chunk is active or not. That's how one could solve this issue, let the user decicde if they want it or not.

**Comment by asiekierka** [comment]
BuildCraft cannot fix it.

The pump *can* consume water. We already have an option added to make pumps consume water. This is something BuildCraft can change, and offers the option to. There is no 9-block limit or anything like that.

However, if you have a large enough pool, Minecraft *itself* will regenerate the water, which means the pump will once again consume the new water source block created by Minecraft itself. As changing the behaviour of Minecraft's water is out of scope for BuildCraft, we cannot fix this. This is why, to get what you want, you need to install a separate mod which makes Minecraft's built-in water finite.

**Comment by PiotrKFtw** [comment]
This was never an issue related to finite water - or whatever fungus bungus was made up in this thread. The issue, during the time when this  thread was made, was that when watching a larger water pool LIVE (height max 2), there were no water sources disappearing (pre-regeneration), thus the option consumeWater not working as advertised, and in older buildcraft versions you COULD see the water source blocks getting removed by the pump, and THEN minecraft regenerating it. Who knows, maybe one of BC devs had fixed it in later version, no idea as I'm not playing mc since a while, but it's nice if it was fixed.

**Comment by asiekierka** [comment]
[Like this?](https://img.asie.pl/5TJX.mp4) Note that we have not changed any code pertaining to the Pump since late 2017.

**Comment by PiotrKFtw** [comment]
Def not. I'd need to jump on minecraft again and record the issue, and additionally record second video which would show how it should work. Which I don't have time for at this moment.

**Comment by asiekierka** [comment]
That's fine, no worries. Let me know if you ever want to unearth this issue, then - until I have more information, then I don't see what is broken here.
